utility/db.js

```diff
@@ -47,7 +47,34 @@ class DatabaseUtility{
     }
 
     dbClose(){
-        this.connection.end();
+        this.connection && this.connection.end();
+    }
+
+    
+    //extract admission_level into an object    
+    getPromiseOfAdmissionLevel()  {
+        var strQueryAdmissionLevelSql = DatabaseUtility.prepare(['name', 'id'],'admission_level');
+        var findData = this.query(strQueryAdmissionLevelSql);
+        return findData;
+    }
+
+    handleGetPromiseOfAdmissionLevel(promise){
+        var self = this;
+        promise.then(function(data){
+            for(var i = 0; i < data.length; i++){
+                var name = data[i]['name'];
```

Comment:
we should make sure data[i] do exist, or access value of data[i] will throw out error immediately